tests/test_metadata_serialization.py

```diff
@@ -0,0 +1,206 @@
+# Copyright New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+""" Unit tests testing tuf/api/metadata.py classes
+serialization and deserialization.
+
+"""
+
+import json
+import sys
+import logging
+import unittest
+import copy
+
+from typing import Any, Dict, Callable, Type
+
+from tests import utils
+
+from tuf.api.metadata import (
+    Root,
+    Snapshot,
+    Timestamp,
+    Targets,
+    Key,
+    Role,
+    MetaFile,
+    TargetFile,
+    Delegations,
+    DelegatedRole,
+)
+
+logger = logging.getLogger(__name__)
+
+# DataSet is only here so type hints can be used:
+# It is a dict of name to test dict
+DataSet = Dict[str, Dict[str, Any]]
```

Comment:
oof, I just suggested using more of this annotation but now that I review the actual content, it doesn't actually make sense...

This type was true in one of my original examples when I still had the test case data as dict... but here (and already in my later examples) the tests case data is a string. In which case a custom type isn't actually useful we could just say `Dict[str, str]` instead, right?

(Would be nice to get mypy running on the test files as well so we didn't have to review these details)

Reply:
I decided to not remove the `DataSet` type because it saved me from splitting an existing line into two. 

I still, updated the definitions for it though.